lib/SILOptimizer/Mandatory/TFDifferentiation.cpp

```diff
@@ -3364,15 +3405,28 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
     getBuilder().setInsertionPoint(adjointEntry);
 
     SmallVector<SILValue, 8> retElts;
-    auto origParamArgs = original.getArgumentsWithoutIndirectResults();
-    for (auto i : task->getIndices().parameters.set_bits()) {
-      auto origParam = origParamArgs[i];
-      auto adjParam = originalParametersInAdj[i];
+    auto origParams = original.getArgumentsWithoutIndirectResults();
+    auto addRetElt = [&](unsigned parameterIndex) -> void {
```

Comment:
Please add a comment to explain what this closure does. It's good to do this for any local closure for readability.